src/sentry/utils/outcomes.py

```diff
@@ -122,13 +115,51 @@ def track_outcome(
         ),
     )
 
-    metrics.incr(
-        "events.outcomes",
-        skip_internal=True,
-        tags={
-            "outcome": outcome.name.lower(),
-            "reason": reason,
-            "category": category.api_name() if category is not None else "null",
-            "topic": topic_name,
-        },
+
+def track_outcome(
+    org_id: int,
+    project_id: int,
+    key_id: Optional[int],
+    outcome: Outcome,
+    reason: Optional[str] = None,
+    timestamp: Optional[datetime] = None,
+    event_id: Optional[str] = None,
+    category: Optional[DataCategory] = None,
+    quantity: Optional[int] = None,
+) -> None:
+    """
+    This is a central point to track org/project counters per incoming event.
+    NB: This should only ever be called once per incoming event, which means
+    it should only be called at the point we know the final outcome for the
+    event (invalid, rate_limited, accepted, discarded, etc.)
+
+    This sends the "outcome" message to Kafka which is used by Snuba to serve
+    data for SnubaTSDB and RedisSnubaTSDB, such as # of rate-limited/filtered
+    events.
+    """
+    return track_outcome_custom_publish(
+        _publish,
+        org_id,
+        project_id,
+        key_id,
+        outcome,
+        reason,
+        timestamp,
+        event_id,
+        category,
+        quantity,
```

Comment:
nit: Because the list of arguments is so long, I would pass these as keyword args (`quantity=quantity`, etc.).